### PR TITLE
discord: enable aarch64-darwin support for discord-canary

### PR DESCRIPTION
**Pull request by Sciencentistguy**
###### Description of changes
Previous comment stated that only `discord-ptb` supported `aarch64-darwin`. This may have been correct at the time, but currently `-canary` supports `aarch64-darwin` as well.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by devins2518 — APPROVED** [review]
Result of `nixpkgs-review pr 179379` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>discord-canary</li>
  </ul>
</details>

**Comment by Sciencentistguy** [comment]
Rebased on master

File: pkgs/applications/networking/instant-messengers/discord/default.nix [hunk 1]
```diff
@@ -1,76 +1,86 @@
...
-    stable = "0.0.264";
-    ptb = "0.0.59";
-    canary = "0.0.283";
-  };
-  version = versions.${branch};
-  srcs = let
-    darwin-ptb = fetchurl {
-      url = "https://dl-ptb.discordapp.net/apps/osx/${version}/DiscordPTB.dmg";
-      sha256 = "sha256-LS7KExVXkOv8O/GrisPMbBxg/pwoDXIOo1dK9wk1yB8=";
+{
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/instant-messengers/discord/default.nix:1]: Please revert this formatting.

**Comment by Sciencentistguy** [comment]
Reverted formatting. Apologies, my editor uses alejandra as a nix formatter by default.

**Comment by SuperSandro2000** [comment]
Please rebase on master

**Comment by Sciencentistguy** [comment]
@SuperSandro2000 done 🙂

**Review by SuperSandro2000 — APPROVED** [review]

**Pull request merged**